extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java

```diff
@@ -275,6 +278,18 @@ void processAnnotationsAndIndexFiles(
         proxyDefinitions
                 .produce(new NativeImageProxyDefinitionBuildItem("com.sun.xml.txw2.TypedXmlWriter"));
 
+        if (jaxbConfig.indexPath.isPresent()) {
+            List<JaxbFileRootBuildItem> fileRootBuildItems = jaxbConfig.indexPath.get().stream()
+                    .map(JaxbFileRootBuildItem::new).collect(Collectors.toList());
+
+            for (JaxbFileRootBuildItem i : fileRootBuildItems) {
+                try (Stream<Path> stream = iterateResources(applicationArchivesBuildItem, i.getFileRoot())) {
+                    stream.filter(p -> p.getFileName().toString().equals("jaxb.index"))
+                            .forEach(p1 -> handleJaxbFile(p1, resource, reflectiveClass));
+                }
+            }
+        }
```

Comment:
This is exactly the same as the code below. Add to a separate List containing all the `JaxbFileRootBuildItem` elements before iterating in the code below?

Reply:
Done.